functions/snippets/pubsub/src/main/java/com/example/functions/pubsub/PublishMessage.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.functions.pubsub;
+
+// [START functions_pubsub_publish]
+import com.google.cloud.functions.HttpFunction;
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.ProjectTopicName;
+import com.google.pubsub.v1.PubsubMessage;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
```

Comment:
We've decided to recommend [sl4fj for logging](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#logging) so we can be consistent everywhere. Would you mind updating to use that instead?

Reply:
Unfortunately `slf4j`'s API itself does **not** work with `TestLogHandler` from `com.google.guava:guava-testlib:28.2-jre`, as it doesn't have `java.util.logging`'s `addHandler` function.  (`TestLogHandler` is used to verify that certain statements were logged.)

Happy to make this change (in another PR if need be) if we can find a workaround. The only one I know of right now is to mock log calls that use `TestLogHandler`. (@eamonnmcmanus had previously voted against this approach though)

Reply:
Sounds good for now, but we should figure out how to handle these in tests or flip back our decision to require it.

@lesv FYI